kolibri/core/utils/cache.py

```diff
@@ -11,3 +13,85 @@ def __get_process_cache():
 
 
 process_cache = SimpleLazyObject(__get_process_cache)
+_namespace_locks = {}
+
+
+class NamespacedCacheProxy(BaseCache):
+    """
+    Namespaces keys and retains a record of inserted keys for easy clearing of
+    all namespaced keys in the cache
+    """
+
+    def __init__(self, cache, namespace, **params):
+        """
+        :type cache: BaseCache
+        :type namespace: str
+        """
+        params.update(KEY_PREFIX=namespace)
+        super(NamespacedCacheProxy, self).__init__(params)
+        self.cache = cache
+        self._lock = _namespace_locks.setdefault(namespace, RWLock())
```

Comment:
These objects are process local - if this is running in a job on e.g. Kolibri-server, then this would be separate from main server process (which is running in uwsgi), and when we finally get iceqube to properly use multiprocessing, then multiple jobs would not respect each other's locks here.

Would this be a problem for the kind of locking you are trying to do here?

Reply:
Ah yeah that's true, so I should use the disk lock class?

Reply:
Yeah, probably best to use that, using the `process_cache`, as with the `db_write_lock`:
https://github.com/learningequality/kolibri/blob/develop/kolibri/core/tasks/utils.py#L175

Reply:
(this way it uses Redis in Redis deploys, but otherwise falls back to disk cache)